numpy/lib/npyio.py

```diff
@@ -733,10 +733,15 @@ def _getconv(dtype):
     """ Find the correct dtype converter. Adapted from matplotlib """
 
     def floatconv(x):
-        x.lower()
-        if '0x' in x:
-            return float.fromhex(x)
-        return float(x)
+        try:
+            return float(x)
+        except ValueError as exc:
+            try:
+                if '0x' in x:
+                    return float.fromhex(x)
+            except ValueError:
+                pass
+            raise exc
```

Comment:
It might be good to add a comment here to explain why the nested try's & ifs.

Reply:
Done.